src/components/network-area-diagram-viewer/metadata-utils.ts

```diff
@@ -333,3 +333,8 @@ export function getEdgeType(edge: EdgeMetadata): EdgeType {
 export function getStringEdgeType(edge: EdgeMetadata): string {
     return EdgeType[getEdgeType(edge)];
 }
+
+export function getBusNodesMetadata(nodeId: string, busNodes: BusNodeMetadata[]): BusNodeMetadata[] {
+    const buses = busNodes.filter((bus) => bus.vlNode === nodeId);
```

Comment:
As we're going through all metadata, it might be worth to avoid filtering for each vlNode to loop over the bus node metadata only once.